util/tracing/span.go

```diff
@@ -0,0 +1,45 @@
+package tracing
+
+import (
+	"os"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+/*
+	Poor Mans OpenTracing.
+
+	Standardizes logging of operation duration.
+*/
+
+var enabled = false
+var logger = log.New()
+
+func init() {
+	enabled = os.Getenv("ARGOCD_TRACING_ENABLED") == "1"
+	logger.Infof("tracing enabled=%v", enabled)
+}
+
+type Span struct {
+	operationName string
+	baggage       map[string]interface{}
+	start         time.Time
+}
+
+func (s Span) Finish() {
+	if enabled {
+		logger.WithFields(s.baggage).
+			WithField("operation_name", s.operationName).
+			WithField("time_ms", time.Since(s.start).Seconds()*1e3).
+			Info()
+	}
+}
+
+func (s Span) SetBaggageItem(key string, value interface{}) {
```

Comment:
Can we use method chaining here similar to log context? E.g.

```
span.WithBaggage('myKey', 'myValue').WithBaggage('myKey1', 'myValue1')
```

Reply:
I deliberately didn't do this as it is meant to API the opentracing API